openmdao/recorders/sqlite_reader.py

```diff
@@ -504,6 +547,8 @@ def _list_cases_recurse_flat(self, coord=None):
             else:
                 raise RuntimeError('Unexpected table name in global iterations:', table)
 
+            self.source_cases_table[table].append(case_coord)
```

Comment:
I think this should be inside the following if statement...  in order to respect the `coord` argument